### DISCUSSION TITLE
Nextjs - How to use middleware.ts

### DISCUSSION THREAD

**Discussion by JumBay** [nrwl/nx#15972] [General]
Hi,

I've created a starter app with Integrated Repos and created a nextjs app using `nx g @nrwl/next:app my-new-app` but I cannot manage to make middlewares work. I've created a `middleware.ts` at the root of my `my-new-app` directory (as explained https://nextjs.org/docs/advanced-features/middleware) but it's liked ignored. 

If I create a monorepo project without nx, the middleware works as supposed.

Here is the repo https://github.com/JumBay/nx-nextjs-middleware

Thanks

**Comment by Morozzzko** [comment]
Hey, I'm facing the same issue. Could you please tell me how you solved it? Thanks!

**Reply by JumBay** [reply]
Actually it was because we set the property `pageextension` in nextjs.config file and it excluded the middleware.
